### PR TITLE
cw_decred: add libdcrwallet dependency and link library for android, ios and macos

### PR DESCRIPTION
**Pull request by itswisdomagain**
Issue Number (if Applicable): Fixes #

# Description
Builds on #1165.

Adds build scripts for building the decred wallet library for android, ios and macos.

# Pull Request - Checklist  

- [x] Initial Manual Tests Passed
- [x] Double check modified code and verify it with the feature/task requirements
- [x] Format code
- [x] Look for code duplication
- [x] Clear naming for variables and methods

### CODE REVIEW

**Review by OmarHatem28 — APPROVED** [review]
Overall looks good, just 2 minor comments

File: scripts/ios/app_config.sh [hunk 1]
```diff
@@ -24,15 +24,15 @@ cp -rf ./ios/Runner/InfoBase.plist ./ios/Runner/Info.plist
...
 
 case $APP_IOS_TYPE in
-        $MONERO_COM)
+    $MONERO_COM)
 		CONFIG_ARGS="--monero"
 		;;
+
         $CAKEWALLET)
-		CONFIG_ARGS="--monero --bitcoin --haven --ethereum --polygon --nano --bitcoinCash"
+		CONFIG_ARGS="--monero --bitcoin --haven --ethereum --polygon --nano --bitcoinCash --decred"
```
[the remark this answers is not in the document]
Comment (itswisdomagain) [reply scripts/ios/app_config.sh:32]: This was already done for android in #1165 ([diff](https://github.com/cake-tech/cake_wallet/pull/1165/files#diff-ad0befc05deca06c72d7dd0959c783bbae102624afe6a7d022fc77ebcc161a0b)).

File: cw_decred/lib/wallet_service.dart [hunk 2]
```diff
@@ -55,13 +69,10 @@ class DecredWalletService extends WalletService<
   @override
   Future<void> rename(
       String currentName, String password, String newName) async {
-    final currentWalletInfo = walletInfoSource.values.firstWhereOrNull(
-        (info) => info.id == WalletBase.idFor(currentName, getType()))!;
-    final currentWallet = await DecredWalletBase.open(
-        password: password, name: currentName, walletInfo: currentWalletInfo);
-
+    final currentWallet = await openWallet(currentName, password);
```
[the remark this answers is not in the document]
Comment (itswisdomagain) [reply cw_decred/lib/wallet_service.dart]: Fixed in https://github.com/cake-tech/cake_wallet/pull/1240/commits/31a081ae76f0117404782d36bc8383adca0634c6.

File: cw_decred/lib/wallet.dart
Comment (OmarHatem28) [inline cw_decred/lib/wallet.dart:1]: when dealing with file system we do expect some weird behaviors from different devices, so it's better to gracefully handle this here, and try other alternatives as well, you can check the `renameWalletFiles` function in `monero_wallet` and you can see what I mean.

non blocker so I will merge anyway, but something to consider adding in future PRs.

**Pull request merged**